hack/rp-dev/rp_funcs.sh

```diff
@@ -0,0 +1,291 @@
+  log "Success step 1 ✅ - Directory '$secrets_dir' has been created."
+
+  export AZURE_PREFIX="$azure_prefix" NO_CACHE="true" ARO_INSTALL_VIA_HIVE="true" ARO_ADOPT_BY_HIVE="true" DATABASE_NAME="ARO"
+  azure_resource_name=${azure_prefix}-aro-$location
+  azure_resource_name="${azure_resource_name:0:20}"
+  export RESOURCEGROUP="$azure_resource_name" DATABASE_ACCOUNT_NAME="$azure_resource_name" KEYVAULT_PREFIX="$azure_resource_name"
+  # TODO chekc if it is needed here
+  export ARO_IMAGE="${azure_prefix}aro.azurecr.io/aro:${git_commit}"
+
+  # Generate new dev-config.yaml
+  if [ -f "dev-config.yaml" ]; then
+    rm dev-config.yaml
+  fi
+  make dev-config.yaml
+  if [ ! -f "dev-config.yaml" ]; then
+    abort "File dev-config.yaml does not exist."
+  fi
+  log "Success step 2 ✅ - Config file dev-config.yaml has been created"
+}
+
+pre_deploy_resources() {
+    err_str="Usage $0 <AZURE_PREFIX> <LOCATION> <RESOURCE_GROUP> [SKIP_DEPLOYMENTS]. Please try again"
+    local azure_prefix=${1?$err_str}
+    local location=${2?$err_str}
+    local rp_resource_group=${3?$err_str}
+    local skip_deployments=${4:-"true"}
+
+    # Don't skip deployment creation when SKIP_DEPLOYMENTS was set to "false" 
+    if is_it_boolean $skip_deployments && ! $skip_deployments; then
+      log "'SKIP_DEPLOYMENTS' env var was set to false. Don't skip predeployment."
+      make pre-deploy-no-aks
+    fi
+  
+    local num_deployment=0
+    resource_groups=("${azure_prefix}-global" "${azure_prefix}-subscription" "${azure_prefix}-gwy-${location}" "${azure_prefix}-gwy-${location}" "$rp_resource_group" "$rp_resource_group")
+    deployments=("rp-global-${location}" "rp-production-subscription-${location}" "gateway-production-predeploy" "gateway-production-managed-identity" "rp-production-managed-identity" "rp-production-predeploy-no-aks")
+    for i in "${!deployments[@]}"; do
+      check_deployment ${resource_groups[i]} ${deployments[i]} && num_deployment=$((num_deployment + 1))
+    done
+    if [[ ${num_deployment} -lt 6 ]]; then
+      log "Deploy predeployment resources prior to AKS. ${num_deployment}/6 deployments have been deployed." 
+      make pre-deploy-no-aks
+    else
+      log "All the 6 deployments exists. ⏩📋 Predeployment was skipped"
+    fi
+    log "Success step 3 ✅ - deploy pre-deployment resources prior to AKS"
+}
+
+add_hive(){
+  err_str="Usage $0 <LOCATION> <RESOURCE_GROUP> [SKIP_DEPLOYMENTS]. Please try again"
+  local location=${1?$err_str}
+  local resource_group=${2?$err_str}
+  local skip_deployments=${3:-"true"}
+
+  is_it_boolean $skip_deployments
+
+  source hack/devtools/deploy-shared-env.sh
+  if $skip_deployments && check_deployment ${resource_group} dev-vpn; then
+    log "⏩📋 VPN deployment was skipped"
+  else
+    deploy_vpn_for_dedicated_rp 
+    log "Success step 4a 🚀 - VPN has been deployed"
+  fi
+
+  if $skip_deployments && check_deployment ${resource_group} aks-development; then
```

Comment:
We can also check with `check_vmss` whether the AKS VMSS exist but I don't understand how the VMSS suffix name is created (maybe random?). For instance VMSS name `aks-systempool-35203733-vmss`